src/kdf.rs

```diff
@@ -29,6 +31,22 @@ pub enum Kdf {
     Sha512 = 3,
 }
 
+impl FromStr for Kdf {
+    type Err = IdLookupError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match &*s.to_lowercase() {
+            #[cfg(feature = "kdf-sha256")]
+            "sha256" | "sha-256" => Ok(Self::Sha256),
```

Comment:
We may want to include the "HKDF-" prefix in these identifiers for completeness and future-proofing. See [RFC9180, Table 3](https://www.rfc-editor.org/rfc/rfc9180.html#table-3).

Reply:
added a second commit (869c6d87a7bb7b3f135b54c02a63bd3f1fb3590b) that handles `hkdf-sha256` and `HkdfSha256`

Reply:
Should we *require* the prefix?

Reply:
I think accepting either is probably fine.